internal/k8s/ako_init.go

```diff
@@ -216,6 +216,74 @@ func (c *AviController) HandleConfigMap(k8sinfo K8sinformers, ctrlCh chan struct
 	return nil
 }
 
+func (c *AviController) AddBootupSecretEventHandler(k8sinfo K8sinformers, stopCh <-chan struct{}, startSyncCh chan struct{}) {
+	cs := k8sinfo.Cs
+	utils.AviLog.Infof("Creating event broadcaster for Avi Secret Handling")
+	eventBroadcaster := record.NewBroadcaster()
+	eventBroadcaster.StartLogging(utils.AviLog.Debugf)
+	eventBroadcaster.StartRecordingToSink(&typedcorev1.EventSinkImpl{Interface: cs.CoreV1().Events("")})
+	NCPSecretHandler := cache.ResourceEventHandlerFuncs{
+		AddFunc: func(obj interface{}) {
+			if lib.AviSecretInitialized {
+				return
+			}
+			if c.ValidAviSecret() {
+				startSyncCh <- struct{}{}
+				startSyncCh = nil
+			}
+		},
+		UpdateFunc: func(old, obj interface{}) {
+			if lib.AviSecretInitialized {
+				return
+			}
+			if c.ValidAviSecret() {
+				startSyncCh <- struct{}{}
+				startSyncCh = nil
+			}
+		},
+	}
+	c.informers.SecretInformer.Informer().AddEventHandler(NCPSecretHandler)
+
+	go c.informers.SecretInformer.Informer().Run(stopCh)
+	if !cache.WaitForCacheSync(stopCh, c.informers.SecretInformer.Informer().HasSynced) {
+		runtime.HandleError(fmt.Errorf("Timed out waiting for caches to sync"))
+	} else {
+		utils.AviLog.Info("Caches synced for Avi Secret informer")
+	}
+}
+
+func (c *AviController) ValidAviSecret() bool {
```

Comment:
Same for this method. Ideally we should create a controller class within the infra package.